### PR TITLE
Expose YARPC option for custom interceptors

### PR DESCRIPTION
**Pull request by sectioneight**
This way our users can add custom YARPC interceptors as an option
in module creation.

GFM-117

### CODE REVIEW

**Comment by sectioneight** [comment]
@uber-go/service-framework

File: modules/rpc/yarpc_options_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	mc := &service.ModuleCreateInfo{
+		Items: map[string]interface{}{
+			_interceptorKey: "foo",
+		},
+	}
+
+	assert.Panics(t, func() {
+		opt(mc)
+	})
+}
```
Comment (glibsm) [inline modules/rpc/yarpc_options_test.go]: maybe add a test for a dummy interceptor in action?

Comment (glibsm) [inline modules/rpc/yarpc_options_test.go]: We only test a `NopInterceptor`. I was asking if we should test with a real interceptor so we can validate it's actually executed and all that (without testing the internals of the yarpc too much)

Comment (sectioneight) [inline modules/rpc/yarpc_options_test.go]: Meh, I don't think there's much value in testing that YARPC interceptors work.

**Pull request merged**